### DISCUSSION TITLE
Schedule definition with 'and' does not work

### DISCUSSION THREAD

**Discussion by stefan-g** [renovatebot/renovate#29128] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

The renovate-runner has created and merge a branch outside the defined schedule time. It seems that 'and' combination is ignored The schedule config in the renovate.json looks like this:
"schedule": ["after 8am and before 12am every weekday"]

In the logs of the renovate-runner the config is merge like this:

```
DEBUG: Repository config (repository=demo)
       "fileName": "renovate.json",
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": [
           "local>renovate/renovate-shared-config:spring-boot-default-config.json"
         ],
         "schedule": ["after 8am and before 12am every weekday"]
       }
DEBUG: Repo is onboarded (repository=demo)
DEBUG: migrateAndValidate() (repository=demo)
DEBUG: Config migration necessary (repository=demo)
       "oldConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": [
           "local>renovate/renovate-shared-config:spring-boot-default-config.json"
         ],
         "schedule": ["after 8am and before 12am every weekday"]
       },
       "newConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": [
           "local>renovate/renovate-shared-config:spring-boot-default-config.json"
         ],
         "schedule": ["after 8am every weekday", "before 12am every weekday"]
       }
DEBUG: Post-massage config (repository=demo)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": [
           "local>renovate/renovate-shared-config:spring-boot-default-config.json"
         ],
         "schedule": ["after 8am every weekday", "before 12am every weekday"]
```

When the schedule time is checked it says that it found a matching time but checks only one criteria

```
DEBUG: Checking 2 schedule(s) (repository=demo, branch=renovate/patch-all-patch-dependencies)
DEBUG: Checking schedule "after 8am every weekday" (repository=demo, branch=renovate/patch-all-patch-dependencies)
       "parsedSchedule": {
         "schedules": [{"t_a": [28800], "d": [2, 3, 4, 5, 6]}],
         "exceptions": [],
         "error": -1
       }
DEBUG: Matches schedule after 8am every weekday (repository=demo, branch=renovate/patch-all-patch-dependencies)
DEBUG: Branch needs creating (repository=demo, branch=renovate/patch-all-patch-dependencies)
```

Is that a bug or does an other way exists to define a time range?

**Comment by rarkins** [comment]
Unfortunately the "later" syntax is quite limited at times, so I recommend you use cron syntax instead, which is very well defined.

For example I fed this into ChatGPT: https://chatgpt.com/share/8387dd3b-3957-4c20-9579-93e1fb394283

It also reveals that you maybe made a mistake in your text anyway? 12am means midnight, I'm guessing you meant 12pm (midday)?

**Reply by stefan-g** [reply]
yes, you are right. i meant 12 pm. will check if that works. thanks

**Reply by rarkins** [reply, marked as answer]
https://chatgpt.com/share/eeca5b2c-79ac-4a55-887a-131250c00997
